at_client/src/main/java/org/atsign/client/util/KeysUtil.java

```diff
@@ -18,68 +18,104 @@
+            // if file does not exist at
+            // ./keys/$atSign_key.atKeys
+            // Create dir to store key files
+            if (!file.exists()) {
+                _makeRootFolder();
+            }
         }
-
         String json = new String(Files.readAllBytes(file.toPath()), StandardCharsets.UTF_8);
-        @SuppressWarnings("unchecked") Map<String, String> encryptedKeys = mapper.readValue(json, Map.class);
+        @SuppressWarnings("unchecked")
+        Map<String, String> encryptedKeys = mapper.readValue(json, Map.class);
 
-        // All the keys are encrypted with the AES self encryption key (which is left unencrypted)
+        // All the keys are encrypted with the AES self encryption key (which is left
+        // unencrypted)
         String selfEncryptionKey = encryptedKeys.get(selfEncryptionKeyName);
 
         Map<String, String> keys = new HashMap<>();
         keys.put(selfEncryptionKeyName, selfEncryptionKey);
-        keys.put(pkamPublicKeyName, EncryptionUtil.aesDecryptFromBase64(encryptedKeys.get(pkamPublicKeyName), selfEncryptionKey));
-        keys.put(pkamPrivateKeyName, EncryptionUtil.aesDecryptFromBase64(encryptedKeys.get(pkamPrivateKeyName), selfEncryptionKey));
-        keys.put(encryptionPublicKeyName, EncryptionUtil.aesDecryptFromBase64(encryptedKeys.get(encryptionPublicKeyName), selfEncryptionKey));
-        keys.put(encryptionPrivateKeyName, EncryptionUtil.aesDecryptFromBase64(encryptedKeys.get(encryptionPrivateKeyName), selfEncryptionKey));
+        keys.put(pkamPublicKeyName,
+                EncryptionUtil.aesDecryptFromBase64(encryptedKeys.get(pkamPublicKeyName), selfEncryptionKey));
+        keys.put(pkamPrivateKeyName,
+                EncryptionUtil.aesDecryptFromBase64(encryptedKeys.get(pkamPrivateKeyName), selfEncryptionKey));
+        keys.put(encryptionPublicKeyName,
+                EncryptionUtil.aesDecryptFromBase64(encryptedKeys.get(encryptionPublicKeyName), selfEncryptionKey));
+        keys.put(encryptionPrivateKeyName,
+                EncryptionUtil.aesDecryptFromBase64(encryptedKeys.get(encryptionPrivateKeyName), selfEncryptionKey));
         return keys;
     }
 
-    private static void _makeRootFolder() throws IOException {
-        Path dir = Paths.get(rootFolder);
-        if (! Files.exists(dir)) {
-            Files.createDirectories(dir);
-        }
+    private static File getKeysFile(AtSign atSign, String rootFolder) {
+        return new File(rootFolder + atSign + keysFileSuffix);
+    }
+
+    // Changing _makeRootFolder to be a last resort func
+    // This creates a dir in ~/
+    private static void _makeRootFolder() throws IOException, AtException {
+        String keysFileName = System.getProperty("user.home") + "/.atsign/keys/";
+        Path rootDir = Paths.get(keysFileName);
+        Files.createDirectories(rootDir);
+        throw new AtException("loadKeys: No file at ~/.atsign/keys or your_proj/keys" +
```

Comment:
Exception shouldn't be thrown here, but should be instead thrown where the code context understands whether to throw an exception or not